operator/pkg/remote/kyma_synchronization_context.go

```diff
@@ -262,26 +265,91 @@ func (c *KymaSynchronizationContext) ReplaceWithVirtualKyma(kyma *operatorv1alph
 		modules[m.Name] = m
 	}
 
-	kyma.Spec.Modules = []operatorv1alpha1.Module{}
+	kyma.Spec.Modules = []v1alpha1.Module{}
 	for _, m := range modules {
 		kyma.Spec.Modules = append(kyma.Spec.Modules, m)
 	}
 }
 
 func (c *KymaSynchronizationContext) EnsureNamespaceExists(ctx context.Context, namespace string) error {
-	ns := &corev1.Namespace{ObjectMeta: v1.ObjectMeta{Name: namespace}}
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
 	var err error
 	if err = c.runtimeClient.Get(ctx, client.ObjectKey{Name: namespace}, ns); k8serrors.IsNotFound(err) {
 		return c.runtimeClient.Create(ctx, ns)
 	}
 	return err
 }
 
-func GetRemoteObjectKey(kyma *operatorv1alpha1.Kyma) client.ObjectKey {
+func GetRemoteObjectKey(kyma *v1alpha1.Kyma) client.ObjectKey {
 	name := kyma.Name
 	namespace := kyma.Namespace
 	if kyma.Spec.Sync.Namespace != "" {
 		namespace = kyma.Spec.Sync.Namespace
 	}
 	return client.ObjectKey{Namespace: namespace, Name: name}
 }
+
+type CatalogSettings struct {
+	Name      string
+	Namespace string
+}
+
+type CatalogEntry struct {
+	Defaults *unstructured.Unstructured `json:"defaults"`
+	Channel  v1alpha1.Channel           `json:"channel"`
+	Target   v1alpha1.Target            `json:"target"`
+	Version  string                     `json:"version"`
+}
+
+func (c *KymaSynchronizationContext) CreateOrUpdateModuleTemplateCatalog(
+	ctx context.Context, catalogSettings CatalogSettings, moduleTemplates *v1alpha1.ModuleTemplateList,
+) error {
+	catalog := &corev1.ConfigMap{}
+	catalog.SetName(catalogSettings.Name)
+	catalog.SetNamespace(catalogSettings.Namespace)
+
+	create := false
+	err := c.runtimeClient.Get(ctx, client.ObjectKeyFromObject(catalog), catalog)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	create = k8serrors.IsNotFound(err)
+
+	if catalog.Data == nil {
+		catalog.Data = make(map[string]string)
+	}
+
+	for i := range moduleTemplates.Items {
+		moduleTemplate := &moduleTemplates.Items[i]
+		moduleName := moduleTemplate.GetLabels()[v1alpha1.ModuleName]
+		var yml []byte
+		var err error
+
+		yml, err = yaml.Marshal(&CatalogEntry{
+			Defaults: &moduleTemplate.Spec.Data,
+			Channel:  moduleTemplate.Spec.Channel,
+			Target:   moduleTemplate.Spec.Target,
+			Version:  moduleTemplate.GetLabels()[v1alpha1.ModuleVersion],
+		})
+
+		if err != nil {
+			return err
+		}
+
+		catalog.Data[moduleName] = string(yml)
+	}
+
+	if create {
+		return c.runtimeClient.Create(ctx, catalog)
+	}
+	return c.runtimeClient.Update(ctx, catalog)
```

Comment:
Is it better we introduce moduleTemplate generation into configmap? Then after first `.Get`, we can compare the generation in catalog and current moduleTemplate, if no change, we don't need to do `.Update` here.

Reply:
ConfigMap has no incrementing generation I think, no?

Reply:
no, I mean save moduleTemplate generation into ConfigMap as additional data, same as what you did for `Target`, `Channel`, then unmarshal it for comparing